File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        i=0
        j=0
        prod=1
        count=0
        while(j<len(nums)):
            prod*=nums[j]
            while(prod>=k and i<=j):
                prod/=nums[i]
                i+=1
            count+=(j-i+1)
            j+=1
        return count

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        int i=0;
        int j=0;
        long long prod=1;
        int count=0;
        while(j<nums.size())
        {
            prod*=nums[j];
            while(prod>=k && i<=j)
            {
                prod/=nums[i];
                i++;
            }
            count+=(j-i+1);
            j++;
        }
        return count;
    }
};